### PR TITLE
multitenant POST /login

### PR DESCRIPTION
**Pull request by mchalski**
depending on the TenantAdmUrl setting, `useradm` is able to work in MT mode, in which it consults the `tenantadm` service to verify the user's tenant. it then issues a token augmented with the `mender.tenant` claim.

Issues: MEN-1063

@maciejmrowiec @mendersoftware/rndity

### CODE REVIEW

**Comment by GregorioDiStefano** [comment]
is this suppose to work standalone? The build failed horribly: https://mender-jenkins.mender.io/job/mender-builder/468/

**Comment by bboozzoo** [comment]
I think you'll need to update this line: https://github.com/mendersoftware/useradm/blob/master/config.go#L40 otherwise it's running multi tenant by default

File: client/tenant/client_tenantadm.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+		conf.Timeout = defaultReqTimeout
+	}
+
+	return &Client{
+		conf: conf,
+	}
+}
+
+func (c *Client) GetTenant(ctx context.Context, username string, client apiclient.HttpRunner) (*Tenant, error) {
+	req, err := http.NewRequest(http.MethodPost,
```
Comment (bboozzoo) [inline client/tenant/client_tenantadm.go]: post?

Comment (mchalski) [inline client/tenant/client_tenantadm.go]: oops, definitely not, will fix

File: client/tenant/client_tenantadm_test.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+			c := NewClient(Config{
+				TenantAdmAddr: s.URL,
+			})
+
+			tenant, err := c.GetTenant(context.Background(), "username", &apiclient.HttpApi{})
+			if tc.err != nil {
+				assert.EqualError(t, err, tc.err.Error())
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, GetTenantsUri, rd.Url.Path)
```
Comment (bboozzoo) [inline client/tenant/client_tenantadm_test.go]: Suppose you should verify HTTP method as well

File: user/useradm.go [hunk 3]
```diff
@@ -51,22 +53,48 @@ type Config struct {
...
+func simpleApiClientGetter() apiclient.HttpRunner {
+	return &apiclient.HttpApi{}
+}
+
 type UserAdm struct {
 	// JWT serialized/deserializer
-	jwtHandler jwt.JWTHandler
-	db         store.DataStore
-	config     Config
+	jwtHandler   jwt.JWTHandler
```
Comment (bboozzoo) [inline user/useradm.go]: mabe you could later rename `jwt.JWTHandler` to simply `jwt.Handler` ?

This should do the trick:
```
gorename -from '"github.com/mendersoftware/useradm/jwt".JWTHandler' -to Handler
gorename -from '"github.com/mendersoftware/useradm/jwt/mocks".JWTHandler' -to Handler
```

File: client/tenant/client_tenantadm.go (hunk 1 above)
Comment (mchalski) [inline client/tenant/client_tenantadm.go]: fixed

File: client/tenant/client_tenantadm_test.go (hunk 2 above)
Comment (mchalski) [inline client/tenant/client_tenantadm_test.go]: fixed

File: user/useradm.go (hunk 3 above)
Comment (mchalski) [inline user/useradm.go]: done

**Comment by mchalski** [comment]
@bboozzoo also fixed the default multitenancy behavior and stacked a rename of `JWTHandler` on top

**Comment by mchalski** [comment]
@bboozzoo oh, and did a refactor of our (un)marshalling. leaving just the Handler (without dedicated Token methods) is what made the most sense for this service.

File: user/useradm_test.go [hunk 4]
```diff
@@ -22,6 +22,9 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/mock"
 
+	"github.com/mendersoftware/go-lib-micro/apiclient"
```
Comment (bboozzoo) [inline user/useradm_test.go]: can you group this import with other 3rd party non-stdlib imports?

Comment (mchalski) [inline user/useradm_test.go]: fixed

**Pull request merged**